src/core_plugins/kibana/public/doc/index.html

```diff
@@ -1,47 +1,75 @@
 <div ng-controller="doc" class="app-container">
+  <div class="kuiLocalNav">
```

Comment:
Thanks for the suggestion! Adding the header in order to unify the appearance with Discover and the context view was the whole point of this PR though (as concluded in https://github.com/elastic/kibana/pull/9198#issuecomment-277694755). I agree that the header is not ideal with its rigid structure, but further optimizations probably make more sense on a Kibana-wide scale.